crates/tauri-build/src/mobile.rs

```diff
@@ -2,11 +2,130 @@
 // SPDX-License-Identifier: Apache-2.0
 // SPDX-License-Identifier: MIT
 
-use std::path::PathBuf;
+use std::{collections::HashSet, path::PathBuf};
 
 use anyhow::{Context, Result};
 use tauri_utils::write_if_changed;
 
+/// Updates the Android manifest to add file association intent filters
+pub fn update_android_manifest_file_associations(
+  associations: &[tauri_utils::config::FileAssociation],
+) -> Result<()> {
+  if associations.is_empty() {
+    return Ok(());
+  }
+
+  let intent_filters = generate_file_association_intent_filters(associations);
+  tauri_utils::build::update_android_manifest("tauri-file-associations", "activity", intent_filters)
+}
+
+fn generate_file_association_intent_filters(
+  associations: &[tauri_utils::config::FileAssociation],
+) -> String {
+  let mut filters = String::new();
+
+  for association in associations {
+    // Get mime types - use explicit mime_type, or infer from extensions
+    let mut mime_types = HashSet::new();
+
+    if let Some(mime_type) = &association.mime_type {
+      mime_types.insert(mime_type.clone());
+    } else {
+      // Infer mime types from extensions
+      for ext in &association.ext {
+        if let Some(mime) = extension_to_mime_type(&ext.0) {
+          mime_types.insert(mime);
+        }
+      }
+    }
+
+    // If we have mime types, create intent filters
+    if !mime_types.is_empty() {
+      for mime_type in &mime_types {
+        filters.push_str("<intent-filter>\n");
```

Comment:
Tauri config has **rank** but this is not being used on Android.
```json
{
  "ext": ["png"],
  "mimeType": "image/png",
  "rank": "Owner"  // ← This exists in config
}
```

Android can specify a _priority_ on intent filters which is a signed integer between -1000 to 1000.

```xml
<intent-filter android:priority="1000">
    <action android:name="android.intent.action.VIEW" />
    <data android:mimeType="image/png" />
</intent-filter>
```